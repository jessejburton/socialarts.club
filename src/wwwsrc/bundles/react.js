import '../react-components/src/index';