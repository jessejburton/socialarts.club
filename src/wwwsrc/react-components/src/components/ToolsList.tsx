import * as React from 'react';

// tslint:disable-next-line:interface-name
interface Tool {
    id: string;
    name: string;
    templateUrlPath: string;
    json: string;
    path: string;
}

// tslint:disable-next-line:interface-name
interface Props {
    Tools: Tool[];
}

const renderTool = (tool: Tool) => {
    const path = `${tool.templateUrlPath}/${tool.id}`;
    return (
        <li key={tool.id}>
            <a href={path}>{tool.name}</a>
        </li>
    );
};

function ToolsList({ Tools }: Props) {
    return (
        <div className='ToolsList'>
            <h2>Tools</h2>
            {Tools.map(renderTool)}
        </div>
    );
}

export default ToolsList;
export { Tool }