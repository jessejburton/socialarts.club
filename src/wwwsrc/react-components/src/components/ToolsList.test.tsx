import * as React from 'react';
import { cleanup, render } from 'react-testing-library';
import ToolsList, { Tool } from './ToolsList';

afterEach(cleanup);

const getData = (length: number) => Array.from(Array(length).keys()).map(i => {
    return {
        id: `id-${i}`,
        json: `json-${i}`,
        name: `name-${i}`,
        path: `path-${i}`,
        templateUrlPath: `templateUrlPath-${i}`,
    } as Tool;
});

test('ToolsList.render should output header', async () => {
    // arrange 
    const expectedLength = 3;
    const tools = getData(expectedLength);

    // act
    const toolsList = render(<ToolsList Tools={tools} />);

    // assert
    const header = toolsList.getByText('Tools');
    expect(header.nodeName).toBe('H2');
    expect(toolsList).toMatchSnapshot();
});

test('ToolsList.render should output correct number of list items', async () => {
    // arrange 
    const count = 3;
    const tools = getData(count);

    // act
    const toolsList = render(<ToolsList Tools={tools} />);

    // assert
    const listItems = toolsList.getAllByText(/^name-/g)
    expect(listItems).toHaveLength(count);
    expect(toolsList).toMatchSnapshot();
});

test('ToolsList.render should use correct href for each named item', async () => {
    // arrange 
    const count = 2;
    const tools = getData(count);

    // act
    const toolsList = render(<ToolsList Tools={tools} />);

    // assert
    for(const tool of tools) {
        const path = `${tool.templateUrlPath}/${tool.id}`;
        const item = toolsList.getByText(tool.name);
        expect(item.nodeName).toBe('A');
        expect(item.getAttribute('href')).toBe(path);
    }
});