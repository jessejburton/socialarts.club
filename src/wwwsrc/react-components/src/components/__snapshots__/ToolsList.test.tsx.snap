// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolsList.render should output correct number of list items 1`] = `
Object {
  "baseElement": <body>
    <div>
      <div
        class="ToolsList"
      >
        <h2>
          Tools
        </h2>
        <li>
          <a
            href="templateUrlPath-0/id-0"
          >
            name-0
          </a>
        </li>
        <li>
          <a
            href="templateUrlPath-1/id-1"
          >
            name-1
          </a>
        </li>
        <li>
          <a
            href="templateUrlPath-2/id-2"
          >
            name-2
          </a>
        </li>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ToolsList"
    >
      <h2>
        Tools
      </h2>
      <li>
        <a
          href="templateUrlPath-0/id-0"
        >
          name-0
        </a>
      </li>
      <li>
        <a
          href="templateUrlPath-1/id-1"
        >
          name-1
        </a>
      </li>
      <li>
        <a
          href="templateUrlPath-2/id-2"
        >
          name-2
        </a>
      </li>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ToolsList.render should output header 1`] = `
Object {
  "baseElement": <body>
    <div>
      <div
        class="ToolsList"
      >
        <h2>
          Tools
        </h2>
        <li>
          <a
            href="templateUrlPath-0/id-0"
          >
            name-0
          </a>
        </li>
        <li>
          <a
            href="templateUrlPath-1/id-1"
          >
            name-1
          </a>
        </li>
        <li>
          <a
            href="templateUrlPath-2/id-2"
          >
            name-2
          </a>
        </li>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ToolsList"
    >
      <h2>
        Tools
      </h2>
      <li>
        <a
          href="templateUrlPath-0/id-0"
        >
          name-0
        </a>
      </li>
      <li>
        <a
          href="templateUrlPath-1/id-1"
        >
          name-1
        </a>
      </li>
      <li>
        <a
          href="templateUrlPath-2/id-2"
        >
          name-2
        </a>
      </li>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
